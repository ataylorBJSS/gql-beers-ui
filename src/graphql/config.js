import ApolloClient from 'apollo-boost'

const client = new ApolloClient({
    uri: 'http://localhost:3000/graphql',
    request: async operation => {
        const token = 'TEST TOKEN'
        operation.setContext({
            headers: {
                authorization: token,
            },
        })
    },
})

export default client
