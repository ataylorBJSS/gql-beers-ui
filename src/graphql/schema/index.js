import gql from 'graphql-tag'

// Example of how we can use fragments
// Removed for now, as there are issue with mocking queries with fragments
// const beerDetails = gql`
//     fragment BeerDetails on Beer {
//         id
//         name
//         tagline
//         image_url
//         abv
//         srm
//         ebc
//     }
// `

export const QRY_BEERS_LIST = gql`
    {
        beers {
            id
            name
            tagline
            image_url
            abv
            srm
            ebc
            ratings {
                total {
                    average
                    count
                }
            }
        }
    }
`

export const QRY_BEER = gql`
    query Beer($id: Int!) {
        beer(id: $id) {
            id
            name
            tagline
            image_url
            abv
            srm
            ebc
            description
            first_brewed
            food_pairing
            ratings {
                total {
                    average
                    count
                }
                rating {
                    rating
                    comment
                }
            }
        }
    }
`

export const ADD_RATING = gql`
    mutation AddRating($parameters: RatingParameters) {
        addRating(parameters: $parameters)
    }
`
