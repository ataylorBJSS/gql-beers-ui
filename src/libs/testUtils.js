import { wait } from 'react-testing-library'

export * from 'react-testing-library'

const waitUntilLoadingIsFinished = (queryByText, loadingText) =>
    wait(() => {
        const isLoading = queryByText(loadingText) != null
        expect(isLoading).toBe(false)
    })

const waitForErrorMessage = (queryByText, ErrText) =>
    wait(() => {
        const hasError = queryByText(ErrText) != null
        expect(hasError).toBe(true)
    })

export { waitUntilLoadingIsFinished, waitForErrorMessage }
