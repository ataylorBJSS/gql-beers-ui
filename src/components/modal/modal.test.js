import React from 'react'
import { Provider } from 'unstated'
import Modal from './modal'
import modalContainer from './modal.state'
import { MockedProvider } from 'react-apollo/test-utils'
import { render, cleanup, fireEvent, waitUntilLoadingIsFinished, waitForErrorMessage } from '../../libs/testUtils'
import { QRY_BEER } from '../../graphql/schema'
import ModalContainer from './modal.state'

const mocks = [
    {
        request: {
            query: QRY_BEER,
            variables: {
                id: '1',
            },
        },
        result: {
            data: {
                beer: {
                    id: '1',
                    name: '__name__',
                    tagline: '__tagline__',
                    image_url: '__IMGURL__',
                    abv: 10,
                    srm: 5,
                    ebc: 5,
                    description: '__DESCRIPTION__',
                    first_brewed: '07/2015',
                    food_pairing: ['__FOOD1__', '__FOOD2__', '__FOOD3__'],
                    ratings: {
                        total: {
                            average: 5,
                            count: 4,
                        },
                        rating: [
                            {
                                rating: 5,
                                comment: 'Nice',
                            },
                        ],
                    },
                },
            },
        },
    },
]

afterEach(cleanup)

it('renders without crashing', () => {
    const modalContainer = new ModalContainer()
    render(
        <MockedProvider mocks={mocks} addTypename={false}>
            <Provider inject={[modalContainer]}>
                <Modal />
            </Provider>
        </MockedProvider>
    )
})

it('renders loading', () => {
    const modalContainer = new ModalContainer()
    const { container } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
            <Provider inject={[modalContainer]}>
                <Modal />
            </Provider>
        </MockedProvider>
    )

    expect(container).toMatchSnapshot()
})

it('renders errors', async () => {
    const modalContainer = new ModalContainer()
    const [data] = mocks
    const mock = { ...data, error: new Error('Error') }
    const { queryByText, container } = render(
        <MockedProvider mocks={[mock]} addTypename={false}>
            <Provider inject={[modalContainer]}>
                <Modal />
            </Provider>
        </MockedProvider>
    )

    await waitForErrorMessage(queryByText, 'Boom!')

    expect(container).toMatchSnapshot()
})

it('renders data', async () => {
    const modalContainer = new ModalContainer()
    modalContainer.openModal('1')
    const { queryByText, container } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
            <Provider inject={[modalContainer]}>
                <Modal />
            </Provider>
        </MockedProvider>
    )

    await waitUntilLoadingIsFinished(queryByText, 'Getting some beer details.....')

    expect(container).toMatchSnapshot()
})

it('shows more details', async () => {
    const modalContainer = new ModalContainer()
    modalContainer.openModal('1')
    const { getByText, queryByText, container } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
            <Provider inject={[modalContainer]}>
                <Modal />
            </Provider>
        </MockedProvider>
    )

    await waitUntilLoadingIsFinished(queryByText, 'Getting some beer details.....')

    fireEvent.click(getByText('More...'))

    expect(container).toMatchSnapshot()
})
