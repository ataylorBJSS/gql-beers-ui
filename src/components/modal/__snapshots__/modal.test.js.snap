// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders data 1`] = `
<div>
  <div>
    <div
      style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; display: flex; justify-content: center; align-items: center; z-index: 10000;"
    >
      <div
        style="transform: translate3d(0, 0, 0); z-index: 10002; width: 50%; background-color: rgb(255, 255, 255); border-radius: 5px; box-sizing: border-box; box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);"
      >
        <div
          class="modalContainer"
        >
          <div
            class="modalHeader"
          >
            <img
              alt="__name__"
              height="250"
              src="__IMGURL__"
              width="80"
            />
            <div
              class="headerDetails"
            >
              <div>
                __name__
                 (abv
                10
                )
              </div>
              <div>
                __tagline__
              </div>
              <div
                class="ratingsWrapper"
              >
                <span
                  style="display: inline-block; direction: ltr;"
                >
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <div
                  class="counter"
                  data-testid="ratings-counter"
                >
                  (
                  4
                  )
                </div>
              </div>
            </div>
          </div>
          <div>
            __DESCRIPTION__
          </div>
          <button>
            More...
          </button>
        </div>
      </div>
      <div
        style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.7); z-index: 10001;"
      />
    </div>
  </div>
</div>
`;

exports[`renders errors 1`] = `
<div>
  <div>
    <div
      style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; display: flex; justify-content: center; align-items: center; z-index: -1;"
    >
      <div
        style="transform: translate3d(0, 100px, 0); opacity: 0; z-index: -1; height: 0px; width: 0px; overflow: hidden;"
      >
        <p>
          Boom!
        </p>
      </div>
      <div
        style="display: none;"
      />
    </div>
  </div>
</div>
`;

exports[`renders loading 1`] = `
<div>
  <div>
    <div
      style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; display: flex; justify-content: center; align-items: center; z-index: -1;"
    >
      <div
        style="transform: translate3d(0, 100px, 0); opacity: 0; z-index: -1; height: 0px; width: 0px; overflow: hidden;"
      >
        <p>
          Getting some beer details.....
        </p>
      </div>
      <div
        style="display: none;"
      />
    </div>
  </div>
</div>
`;

exports[`shows more details 1`] = `
<div>
  <div>
    <div
      style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; display: flex; justify-content: center; align-items: center; z-index: 10000;"
    >
      <div
        style="transform: translate3d(0, 0, 0); z-index: 10002; width: 50%; background-color: rgb(255, 255, 255); border-radius: 5px; box-sizing: border-box; box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);"
      >
        <div
          class="modalContainer"
        >
          <div
            class="modalHeader"
          >
            <img
              alt="__name__"
              height="250"
              src="__IMGURL__"
              width="80"
            />
            <div
              class="headerDetails"
            >
              <div>
                __name__
                 (abv
                10
                )
              </div>
              <div>
                __tagline__
              </div>
              <div
                class="ratingsWrapper"
              >
                <span
                  style="display: inline-block; direction: ltr;"
                >
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    style="cursor: pointer; display: inline-block; position: relative;"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-star fa-w-18 empty"
                      data-icon="star"
                      data-prefix="fas"
                      role="img"
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="display: inline-block; position: absolute; overflow: hidden; top: 0px; left: 0px; width: 100%;"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-star fa-w-18 full"
                        data-icon="star"
                        data-prefix="fas"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
                <div
                  class="counter"
                  data-testid="ratings-counter"
                >
                  (
                  4
                  )
                </div>
              </div>
            </div>
          </div>
          <div>
            __DESCRIPTION__
          </div>
          <button>
            More...
          </button>
        </div>
      </div>
      <div
        style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.7); z-index: 10001;"
      />
    </div>
  </div>
</div>
`;
